units/systemd-time-wait-sync.service.in

```diff
@@ -0,0 +1,23 @@
+#  SPDX-License-Identifier: LGPL-2.1+
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Wait Until Kernel Time Synchronized
+Documentation=man:systemd-time-wait-sync.service(8)
+DefaultDependencies=no
+Before=time-sync.target
+Wants=time-sync.target
+
+[Service]
+Type=oneshot
+ExecStart=!!@rootlibexecdir@/systemd-time-wait-sync
+TimeoutStartSec=infinity
```

Comment:
please add RemainAfterExit=yes. That way the service will stick around after the time was synchronized. This has the benefit that systemd-bootchar and systemd-analyze will consider the service when doing their time measurements, and can show the price one pays for enabling the service.